alonzo/impl/src/Cardano/Ledger/Alonzo/Rules/Utxos.hs

```diff
@@ -33,18 +34,31 @@ import GHC.Records (HasField (..))
 import NoThunks.Class (NoThunks)
 import Shelley.Spec.Ledger.BaseTypes (ShelleyBase, StrictMaybe (..), strictMaybeToMaybe)
 import Shelley.Spec.Ledger.Coin (Coin)
+import Shelley.Spec.Ledger.Credential (Credential)
+import Shelley.Spec.Ledger.Keys (GenDelegs, KeyHash, KeyRole (..))
 import Shelley.Spec.Ledger.LedgerState
 import qualified Shelley.Spec.Ledger.LedgerState as Shelley
 import Shelley.Spec.Ledger.PParams (Update)
 import Shelley.Spec.Ledger.STS.Ppup (PPUP, PPUPEnv (..), PpupPredicateFailure)
-import Shelley.Spec.Ledger.STS.Utxo (UtxoEnv (..))
-import Shelley.Spec.Ledger.TxBody (DCert, TxIn (..), Wdrl)
+import Shelley.Spec.Ledger.Slot (SlotNo)
+import Shelley.Spec.Ledger.TxBody (DCert, PoolParams, Ptr, TxIn (..), Wdrl)
 import Shelley.Spec.Ledger.UTxO (balance, totalDeposits)
 
 --------------------------------------------------------------------------------
 -- The UTXOS transition system
 --------------------------------------------------------------------------------
 
+-- | The UTXO Environment in the Alonzo Era
+data UtxoEnv era = UtxoEnv
+  { slotUE :: SlotNo,
```

Comment:
This postfix type descriptor is unusual, and I don't think we use it anywhere else. So would tend to prefer e.g. `ueSlot`